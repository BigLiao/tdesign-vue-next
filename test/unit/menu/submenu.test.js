import { mount } from '@vue/test-utils';
import { Submenu } from '@/src/menu/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Submenu', () => {
  // test props api
  describe('props', () => {
    it(':name', () => {
      const wrapper = mount({
        render() {
          return <Submenu name={'1'}></Submenu>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':disabled', () => {
      const wrapper = mount({
        render() {
          return <Submenu disabled={true}></Submenu>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('slot', () => {
    it('<icon>', () => {
      const wrapper = mount(Submenu, {
        scopedSlots: {
          icon: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('<default>', () => {
      const wrapper = mount(Submenu, {
        scopedSlots: {
          default: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('<title>', () => {
      const wrapper = mount(Submenu, {
        scopedSlots: {
          title: '<div></div>',
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
