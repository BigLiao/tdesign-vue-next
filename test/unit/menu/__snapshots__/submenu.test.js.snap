// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu props :disabled 1`] = `
<ul class="t-submenu t-is-disabled">
  <!---->
  <!---->
</ul>
`;

exports[`Submenu props :name 1`] = `
<ul class="t-submenu">
  <!---->
  <!---->
</ul>
`;

exports[`Submenu slot <default> 1`] = `
<ul class="t-submenu">
  <!---->
  <!---->
</ul>
`;

exports[`Submenu slot <icon> 1`] = `
<ul class="t-submenu">
  <!---->
  <!---->
</ul>
`;

exports[`Submenu slot <title> 1`] = `
<ul class="t-submenu">
  <!---->
  <!---->
</ul>
`;
